staging/src/k8s.io/apimachinery/pkg/runtime/serializer/codec_factory.go

```diff
@@ -47,30 +47,44 @@ type serializerType struct {
 	StreamSerializer runtime.Serializer
 }
 
-func newSerializersForScheme(scheme *runtime.Scheme, mf json.MetaFactory) []serializerType {
-	jsonSerializer := json.NewSerializer(mf, scheme, scheme, false)
-	jsonPrettySerializer := json.NewSerializer(mf, scheme, scheme, true)
-	yamlSerializer := json.NewYAMLSerializer(mf, scheme, scheme)
+func newSerializersForScheme(scheme *runtime.Scheme, mf json.MetaFactory, options CodecFactoryOptions) []serializerType {
+	jsonSerializer := json.NewSerializerWithOptions(
+		mf, scheme, scheme,
+		json.SerializerOptions{Yaml: false, Pretty: false, Strict: options.Strict},
+	)
+	jsonSerializerType := serializerType{
+		AcceptContentTypes: []string{runtime.ContentTypeJSON},
+		ContentType:        runtime.ContentTypeJSON,
+		FileExtensions:     []string{"json"},
+		EncodesAsText:      true,
+		Serializer:         jsonSerializer,
+
+		Framer:           json.Framer,
+		StreamSerializer: jsonSerializer,
+	}
+	if options.Pretty {
+		jsonPrettySerializer := json.NewSerializerWithOptions(
+			mf, scheme, scheme,
+			json.SerializerOptions{Yaml: false, Pretty: true, Strict: options.Strict},
+		)
+		jsonSerializerType.PrettySerializer = jsonPrettySerializer
+	}
+
+	yamlSerializer := json.NewSerializerWithOptions(
+		mf, scheme, scheme,
+		json.SerializerOptions{Yaml: true, Pretty: false, Strict: options.Strict},
```

Comment:
Is it better to omit the pretty field here since it no-opts (according to `SerializerOptions` documentation)?

Reply:
My thinking here was that Pretty would be set to `false` to imply that it doesn't matter for YAML.
This is the trouble with implementing a single Options struct in Serializer... adding more doesn't sound better though.

I'm fine with omitting this if you think it's more appropriate.

Reply:
Also, @luxas suggests that we use option mutator funcs for `NewSerializer`, but the signature change will require a code refactor, so I'm not sure if it's worth doing.

I'm adding a patch to this PR to try out the mutator funcs for `NewCodecFactory` since the current signature does not have any options.